textAdventure.js

```diff
@@ -0,0 +1,229 @@
+
+function ask(questionText) {
+  return new Promise((resolve, reject) => {
+    rl.question(questionText, resolve);
+  });
+}
+
+start();
+
+async function start(){
+    console.log('\nValid commands:\n"north", "east","south","west"\n"pick up {item}"\n"use {item}"\n"i" for inventory\n"examine {item}"\n"drop {item}"\n"r" for room items\n(type "exit" at any time to leave game,\n or "c" to view controls)\n \n Hello, traveler!');
+    let name= await ask("Enter your name to continue on your path, should you so choose. \n>_");
+    player.name=name;
+    console.log(
+ `\nGreetings, ${player.name}. \n
+You find yourself 
+at the base of the temple.
+To the north, a verdant meadow. The temple 
+doors to the south, a stream in the east, and
+woodland in the distant west.  Better get to 
+grandma's before that pie gets cold.  Where was 
+her place again?`)
+action()
+} 
+
+async function action(){
+    let choice = await ask(`\nwhat would you like to do? \n >_`)
+    choice= choice.toLowerCase();
+
+    if (choice==='east'&& currentRoom.description==='A troll guards the\nbridge to the east;\nover it, a massive bridge.\n"NONE SHALL PASS," it screams'){
+        console.log('The troll gobbled you up!\nif only you had USED something to protect yourself...\n\n*GAME OVER*')
+        process.exit();
+    }   
+    if (choice ==='north'|| choice=== 'east' || choice === 'south' || choice ==='west'){
+        enterRoom(choice);
+
+    }else if(choice==='i'){
+    for(let n in player.inventory){
+        console.log(player.inventory[n].name);
+    }
+}   else if(choice==='r'){
+        console.log(currentRoom.items)
+    }else if(choice==='c'){
+         console.log('Valid commands:\n"north", "east","south","west"\n"pick up {item}"\n"use {item}"\n"i" for inventory\n"examine {item}"\n"drop {item}"\n"r" for room items\n(type "exit" at any time to leave game,\nor "c" to view controls)\n')
+    }else if(choice.includes('examine')){
+         examine(hasItem[choice.replace('examine ','')])
+    }else if(choice.includes('drop')){
+        drop(hasItem[choice.replace('drop ','')])
+
+
+    }else if (choice ==='pick up holy symbol'){ 
+        pickup('holy symbol');
+        insideTemple.description= 'the temple is dimly lit, and quiet;\nbefore you, an altar.\nthere is a spot where the holy symbol was' //pet the bear dammit!, over the river=gramma's pie= win condition!
+    } else if(choice==='pick up sword'){
+        pickup('sword')
+        woodsClearing.description= 'There is a tree stump with a large gash in it.'
+    }else if(choice==='pick up torch'& temple.items.length>1 && currentRoom.name===temple.name){
+            pickup('torch')
+            temple.description= 'a cold stone temple.  there is one torch left'
+    }else if(choice==='pick up torch'& temple.items.length===1 && currentRoom.name===temple.name){
+                pickup('torch')
+                temple.description= 'a cold stone temple.'
```

Comment:
You could nest these in the `else if (choice.includes('pick up'))` logic block to remove some repetition in your code. e.g. 
```js
                else if (choice.includes('pick up')) {
                    if(choice.includes('sword') {
                        pickup('sword`)
                        ...
```
This would help keep related functions bundled together, and make your code a little cleaner.  If you made your acceptable inputs more flexible you could shorten this up even more.